spark_log_parser/loaders/s3.py

```diff
@@ -54,18 +54,25 @@ def load_item(self, filepath):
             if total_size > 20000000000:
                 raise AssertionError(f"Size limit exceeded while downloading from {filepath}.")
 
-        file_streams = []
+        responses: list[StreamingBody] = []
+        file_streams: list[FileChunkStreamWrapper] = []
         # TODO - this serially fetches all matching objects in the bucket at the moment...
         #  There is likely a better way to do this? It may require some ThreadExecutors, though
         for content in contents_to_fetch:
+            data: StreamingBody = self.s3.get_object(Bucket=bucket, Key=content["Key"])["Body"]
```

Comment:
This could be used as context manager too, but looks like it'd require a bit of a refactor...

Reply:
Yeah, a little trickier here since we support reading S3 directories